dns/bind: allow to specify primaryip's port for secondary zone

loopway opened the pull request:
Closes: #4444 

Notes on proposed solution

Bind/Domain.xml:
- change field type to CSVListField, adopted from net/haproxy HAProxy.xml to allow port specification

Bind/named.conf
- adjust jinja2 template instead of introducing new fields to avoid have to deal with config migrations of previous opnsense versions
- if specifying a port for a ipv6 address the following notion has to be used: [address]:port (https://en.wikipedia.org/wiki/IPv6#Address_representation)
- using negative lookbehind in regular expression to avoid matching ipv6 addresses without port
- using multiple lookbehind regular expressions since quantifiers are not allowed and ipv6 addresses can be shortened

fichtner commented on dns/bind/src/opnsense/mvc/app/models/OPNsense/Bind/Domain.xml:
```diff
@@ -17,9 +17,9 @@
-                <primaryip type="NetworkField">
-                    <FieldSeparator>,</FieldSeparator>
-                    <asList>Y</asList>
+                <primaryip type="CSVListField">
```
As far as I know we have an address/port field typ nowadays.

loopway replied:
Found a `IPPortField` field type. This type would require to specify a port and would therefore invalidate address only entries from previous versions...

fichtner replied:
Maybe it needs to be extended. My point is no more CSV magic regarding this.

loopway replied:
I see your point, but I would argue that should be a separate issue/task, requiring changes in core. A quick github search for `type="CSVListField"` shows matches in three files in core and 29 files in plugins. Can we move this forward in the meantime? Changing it to an extended `IPPortField` later should be easy enough. What do you think?

fichtner replied:
I’m only saying I’m not at the office until tomorrow to do core changes. Copy+paste is not the best idea here in 2025 😊

loopway replied:
Any updates on this? By now you must have arrived in the office. 😉

zndrr replied:
@fichtner any chance you could take another look at this? Would love to see it implemented. TIA

fichtner commented:
As discussed... https://github.com/opnsense/core/commit/153d3cc267

zndrr commented:
@loopway do you know if you'll get time to re-evaluate your PR?

loopway commented:
I'll be happy to re-evaluate my PR. I would just need a hint from @fichtner what to use instead of the regex_replace (https://github.com/opnsense/plugins/issues/4444#issuecomment-3112540194).

loopway commented:
I've changed the field type to `IPPortField` and rebased my branch.

loopway commented:
I've put the regex_replace within a local variable and rebased my branch. Added same changes for `forwardserver` as requested by @phprus. 

happy new year! 🍾
